core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/debug/Debugger.java

```diff
@@ -13,49 +13,46 @@
 import org.eclipse.che.api.promises.client.Promise;
 import org.eclipse.che.ide.api.project.tree.VirtualFile;
 
-import com.google.gwt.user.client.rpc.AsyncCallback;
-
+import javax.validation.constraints.NotNull;
 import java.util.List;
 
 /**
  * The general class which provides to manage breakpoints on server.
  *
- * @author <a href="mailto:aplotnikov@codenvy.com">Andrey Plotnikov</a>
+ * @author Andrey Plotnikov
+ * @author Anatoliy Bazko
  */
-public interface Debugger {
+public interface Debugger<T extends DebuggerConnectionContext> extends DebuggerObservable {
+
     /**
      * Adds new breakpoint on server.
      *
      * @param file
      * @param lineNumber
-     * @param callback
      */
-    void addBreakpoint(VirtualFile file, int lineNumber, AsyncCallback<Breakpoint> callback);
+    void addBreakpoint(VirtualFile file, int lineNumber);
 
     /**
      * Deletes breakpoint on server.
      *
      * @param file
      * @param lineNumber
-     * @param callback
      */
-    void deleteBreakpoint(VirtualFile file, int lineNumber, AsyncCallback<Void> callback);
+    void deleteBreakpoint(VirtualFile file, int lineNumber);
 
     /**
-     * Deletes all breakpoints
+     * Deletes all breakpoints on server.
      */
     void deleteAllBreakpoints();
 
     /**
-     * Attaches debugger using given port and host.
-     *
-     * @param host host to which debugger will be connected
-     * @param port port to which debugger will be connected in specified host
+     * Attaches debugger.
      */
-    void attachDebugger(final String host, final int port);
+    Promise<Void> attachDebugger(@NotNull T connectionContext);
```

Comment:
-1
As for me it is more usefull to describe the method's arguments in javadoc and describe possible exceptions which you can get if you pass wrong argument. Using `@NotNull` allows to pass null value, but you don't know what you'll get then.